views/static/lib/js/tally.js

```diff
@@ -0,0 +1,166 @@
+var socket = null;
+var Devices = [];
+var selectedDeviceId = null;
+var noSleep = new NoSleep();
+var KeepAwake = true;
+var device_states = [];
+var mode_preview = false;
+var mode_program = false;
+
+window.onload = onLoad;
+
+function onLoad() {
+  socket = io.connect();
+  socket.on('connect', function () {
+    //connected, let's get some data
+    socket.emit('devices');
+    socket.emit('bus_options');
+  });
+  socket.on('devices', function (deviceArray) {
+    //Returns a list of available Devices for the dropdown list
+    Devices = deviceArray;
+    updateDeviceList();
+  });
+  socket.on('bus_options', function (busOptionsArray) {
+    //Returns a list of available bus options (preview, program, etc.)
+    Bus_Options = busOptionsArray;
+  });
+  socket.on('device_states', function (tallyDataArray) {
+    //process the data received and determine if it's in preview or program and color the screen accordingly
+    device_states = tallyDataArray;
+    ProcessTallyData();
+  });
+  socket.on('flash', function () {
+    //flashes the screen to get the user's attention
+    document.body.className = 'flash';
+    setTimeout(function () {
+      document.body.classList.remove('flash');
+    }, 3000);
+  });
+  socket.on('reassign', function (oldDeviceId, deviceId) {
+    //processes a reassign request that comes from the Settings GUI and relays the command so it originates from this socket
+    socket.emit('listener_reassign', oldDeviceId, deviceId);
+    selectedDeviceId = deviceId;
+    updateTallyInfo();
+  });
+}
+
+function updateDeviceList() {
+  var selDeviceList = document.getElementById('selDeviceList');
+  selDeviceList.options.length = 0;
+  if (Devices.length > 1) { //build a dropdown list of available devices
+    let default_opt = document.createElement('option');
+    default_opt.value = '0';
+    default_opt.text = '(choose your device)';
+    selDeviceList.appendChild(default_opt);
+    for (let i = 0; i < Devices.length; i++) {
+      let opt = document.createElement('option');
+      opt.value = Devices[i].id;
+      opt.text = Devices[i].name;
+      selDeviceList.appendChild(opt);
+    }
+    selDeviceList.setAttribute('onchange', 'selectDeviceFromList();');
+    document.getElementById('divSelectDevice');
+  }
+  else if (Devices.length === 1) { // just load the only one available
+    selectDevice(Devices[0].id);
+  }
+  else {
+    //no devices are available
+    document.getElementById('container').innerHTML = 'No devices are available for tally monitoring at this time.';
+  }
+}
+
+function selectDeviceFromList() {
+  let sel = document.getElementById('selDeviceList');
+  let id = sel.options[sel.selectedIndex].value;
+  if (id !== '0') {
+    selectDevice(id);
+  }
+  KeepScreenAwake(true); //keeps the phone from falling asleep
+}
+
+function selectDevice(deviceId) {
+  socket.emit('device_listen', deviceId, 'web');
+  selectedDeviceId = deviceId;
+  document.getElementById('divSelectDevice');
```

Comment:
This was originally present to hide the div showing the drop down list of devices. I noticed that the drop down list does not disappear anymore, so this needs fixing.

Reply:
Is fixed in the latest push